docker/init_scripts/init

```diff
@@ -8,26 +8,121 @@ shopt -s inherit_errexit # inherit errexit
 # use virtualenvs
 source /backend/bin/activate
 
-# switch to backend directory
-cd /backend || { echo "/backend directory doesn't seem to exist"; exit 1; }
+# Set redis to false if not set by docker env
+ENABLE_EXPERIMENTAL_REDIS="${ENABLE_EXPERIMENTAL_REDIS:="false"}"
+
+# Set INIT_DEBUG to false if not set by docker env
+# (this env var is currently undocumented and usually just needed for development purposes)
+INIT_DEBUG="${INIT_DEBUG:="false"}"
+
+# print debug log output if enabled
+debug_log () {
+    if [ "${INIT_DEBUG}" == "true" ]; then
+        echo "DEBUG:    [init][$(date +"%Y-%m-%d %T")]" "${@}"
+    fi
+}
+
+# print debug log output if enabled
+info_log () {
+    echo "INFO:     [init][$(date +"%Y-%m-%d %T")]" "${@}"
+}
+
+# print debug log output if enabled
+error_log () {
+    echo "ERROR:    [init][$(date +"%Y-%m-%d %T")]" "${@}"
+    exit 1
+}
 
-# Run migrations and start uvicorn
-/init_back &
- 
-# Start nginx
-/init_front &
+# function that runs or main process and creates a corresponding PID file,
+# sadly uvicorn can not do that itself
+start_bin_uvicorn () {
+    # Commands to start our main application and store its PID to check for crashes
+    info_log "starting uvicorn"
+    uvicorn main:app --proxy-headers --host 0.0.0.0 --port 5000 --uds /tmp/uvicorn.sock --workers 2 &
+    UVICORN_PID=$!
+    echo $UVICORN_PID > /tmp/uvicorn.pid
+}
+
+# Commands to start nginx (handling PID creation internally)
+start_bin_nginx () {
+    info_log "starting nginx"
+    if [ "$EUID" -ne 0 ]; then
+        nginx
+    else
+        # if container runs as root, drop permissions
+        nginx -g 'user romm;'
+    fi
+}
+
+# function that runs our independent python scripts and creates corresponding PID files,
+start_python () {
+    SCRIPT="${1}"
+    info_log "starting ${SCRIPT}.py"
+    python3 "${SCRIPT}.py" &
+    WATCHER_PID=$!
+    echo $WATCHER_PID > "/tmp/${SCRIPT}.pid"
+}
+
+watchdog_process_pid () {
+    TYPE=$1
+    PROCESS=$2
+    if [ -f "/tmp/${PROCESS}.pid" ]; then
+        # check if the pid we last wrote to our state file is actually active
+        if [ -d "/proc/$(cat "/tmp/${PROCESS}.pid")" ]; then
+            debug_log "${PROCESS} still running, no need to start"
+        else
+            if [ "${TYPE}" == "bin" ]; then
+                start_bin_"${PROCESS}"
+            elif [ "${TYPE}" == "python" ]; then
+                start_python "${PROCESS}"
+            fi
+        fi
+    else
+            if [ "${TYPE}" == "bin" ]; then
+                start_bin_"${PROCESS}"
+            elif [ "${TYPE}" == "python" ]; then
+                start_python "${PROCESS}"
+            fi
+    fi
+}
+
+# switch to backend directory
+cd /backend || { error_log "/backend directory doesn't seem to exist"; }
```

Comment:
can you confirm it's `/backend` and not `/back`?

Reply:
jep, it was rewritten in my dockerfile refactor to be more readable